src/main/java/com/avengers/gamera/entity/Article.java

```diff
@@ -0,0 +1,65 @@
+package com.avengers.gamera.entity;
+
+import com.avengers.gamera.constant.ArticleType;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import javax.persistence.*;
+import java.time.OffsetDateTime;
+
+@Entity
+@Table(name = "article")
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class Article {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private String coverImgUrl;
+
+    @Column(nullable = false)
+    private String title;
+
+    @Column(nullable = false)
+    private String text;
+
+    @Basic
+    @Column(nullable = false)
+    private String type;
+
+    @Transient
+    private ArticleType articleType;
+
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+
+    @CreationTimestamp
+    private OffsetDateTime createdTime;
+
+    @UpdateTimestamp
+    private OffsetDateTime updatedTime;
+
+    @PostLoad
+    void fillTransient() {
+        if (type != null) {
+            this.articleType = ArticleType.of(type);
+        }
+    }
+
+    @PrePersist
+    void fillPersistent() {
+        if (articleType != null) {
+            this.type = articleType.getTypeName();
+        }
+    }
```

Comment:
I don't understand why we need these methods for extra handling. Spring Data JPA can map Enum to String automatically, just need to add annotation @Enumerated to eum attribute.

Reply:
This is for reference later.  The @Enumerated only works for simply Enum, it cannot map custom enum. 
e.g.   "Nintendo DS", "Sony PlayStation" can't use as enum fields, we have to use NINTENDO_DS( "Nintendo DS"),SONY_PLAYSTATION("Sony PlayStation"). In such case, @Enumerated not working.

Reply:
fixed.   move this  logic to frontend